src/features/web-compat.js

```diff
@@ -11,7 +11,51 @@ function windowSizingFix () {
     window.outerWidth = window.innerWidth
 }
 
+const MSG_WEB_SHARE = 'webShare'
+
+function canShare (data) {
+    if (typeof data !== 'object') return false
+    if (!('url' in data) && !('title' in data) && !('text' in data)) return false // At least one of these is required
+    if ('files' in data) return false // Not supported at the moment
+    if ('title' in data && typeof data.title !== 'string') return false
+    if ('text' in data && typeof data.text !== 'string') return false
+    if ('url' in data) {
+        if (typeof data.url !== 'string') return false
+        try {
+            const url = new URL(data.url, location.href)
+            if (url.protocol !== 'http:' && url.protocol !== 'https:') return false
+        } catch (err) {
+            return false
+        }
+    }
+    if (window !== window.top) return false // Not supported in iframes
+    return true
+}
+
+/** @returns {ShareRequestData} */
+function cleanShareData (data) {
```

Comment:
Can we explain why we're doing this? Is it to match what Android can support?

Reply:
Yeah, this just minimizes checks on the android side. Also some data manipulation is needed to make it more similar to the Chrome behaviour. For example, concatenating url and text if both are present. It's still not 100% the same as the sharesheet UI differs per device.